modules/public/views.py

```diff
@@ -284,6 +367,40 @@ def external(attrs, new=False):
 
         formatted_paragraphs.append(paragraph.replace('\n', '<br />'))
 
+    country_data = []
+    try:
+        metrics_query_json = [
+            {
+                "metric_name": "installed_base_by_country_percent",
+                "snap_id": details['snap_id'],
+                "start": week_ago.strftime('%Y-%m-%d'),
+                "end": today.strftime('%Y-%m-%d')
+            }
+        ]
+
+        metrics_response = api.get_public_metrics(
+            snap_name,
+            metrics_query_json
+        )
+
+        users_by_country = normalize_metrics(
+            metrics_response[0]['series']
+        )
+        country_data = build_country_info(users_by_country)
+    except api.InvalidResponseContent as invalid_response_content:
+        message = str(invalid_response_content)
+        errors = [{'code': 'json-error', 'message': message}]
+    except api.ApiErrorResponse as api_error_exception:
+        if api_error_exception.errors:
+            errors = api_error_exception.errors
+        else:
+            errors = [
+                {
+                    'code': 'unknown-error',
```

Comment:
Again, what is the use of these `code`s? Might this not be over-complicating things? We should only pass through to the template what the template needs to display, and I suspect all it needs is a message.